packages/ui/src/components/Visualization/ContextToolbar/NewEntity/NewEntity.tsx

```diff
@@ -26,7 +47,11 @@ export const NewEntity: FunctionComponent = () => {
        * supported
        */
       const newId = camelResource.addNewEntity(entityType as EntityType);
-      visibleFlowsContext.visualFlowsApi.hideAllFlows();
+      setTimeout(() => {
+        const result = findRootVisualization(controller.getGraph(), newId);
+        controller.fireEvent(SELECTION_EVENT, [result?.id]);
+      }, 300);
```

Comment:
@tplevko we now have the `getVisualizationNodesFromGraph` function to look for canvas nodes.

```ts
      setTimeout(() => {
        const result = getVisualizationNodesFromGraph(
          controller.getGraph(),
          (vizNode) => vizNode.data.entity?.id === newId,
        );

        controller.fireEvent(SELECTION_EVENT, [result[0]?.id]);
      }, 300);
```

That being said, I don't know a better way to ensure the selection happens after the rerender :cry: , so probably what's gonna happen for environments like DevSpaces, this might fail from time to time, since the `SELECTION_EVENT` could happen before the actual rerender.

Would you be open to moving forward without the setTimeout for now until we get a better way to ensure the entity is properly selected?